testsuite/test_config_network.py

```diff
@@ -0,0 +1,223 @@
+  "port": 222
+}
+
+SSH_BROKEN = {
+  "enabled": "whatever",
+  "authorized_keys": [],
+  "port": 222
+}
+
+
+REGISTRATION_DEFAULT = {
+    "username": "admin",
+    "password": "",
+    "domain": "liquidnode.liquid",
+    "lan": {
+        "ip": "10.0.0.1",
+        "netmask": "255.255.255.0",
+        "dhcp_range": "10.0.0.100-255",
+        "hotspot": {
+            "ssid": "",
+            "password": ""
+        },
+        "eth": False
+    },
+    "wan": {
+        "wifi": {
+            "ssid": "",
+            "password": ""
+        }
+    },
+    "ssh": {
+        "enabled": False,
+        "authorized_keys": [],
+        "port": 22
+    }
+}
+
+REGISTRATION1 =  {
+    "username": "johnDoe",
+    "password": "123b456x",
+    "domain": "liquidnode.liquid",
+    "lan": LAN1,
+    "wan": WAN1,
+    "ssh": SSH1
+}
+
+REGISTRATION2 =  {
+    "username": "best_admin_user_ever",
+    "password": "super secret password no one will ever guess",
+    "domain": "mynode.liquidnode.liquid.node.com",
+    "lan": LAN2,
+    "wan": WAN2,
+    "ssh": SSH2
+}
+
+@pytest.fixture
+def client():
+    return APIClient()
+
+def check_get_response(client, endpoint, obj):
+    get = client.get(endpoint)
+    assert obj == get.json()
+    assert 200 == get.status_code
+
+def test_initial_registration_default(client):
+    get = client.get('/api/registration/')
+    assert 200 == get.status_code
+    assert REGISTRATION_DEFAULT == get.json()
+
+@pytest.mark.parametrize("data", [
+    REGISTRATION1,
+    REGISTRATION2,
+])
+def test_initial_registration_setup(client, data):
+    post = client.post('/api/registration/', data=data, format='json')
+    assert 200 == post.status_code
+
+    # try to login
+    assert client.login(username=data['username'], password=data['password'])
+
+    # check the posted configs
+    check_get_response(client, '/api/network/lan/', data['lan'])
+    check_get_response(client, '/api/network/wan/', data['wan'])
+    check_get_response(client, '/api/network/ssh/', data['ssh'])
+
+    # post-registration, all calls to /api/registration should fail
+    get = client.get('/api/registration/')
+    assert 400 == get.status_code
+    assert {'detail': "Registration already done"} == get.json()
+
+    post = client.post('/api/registration/', data=data, format='json')
+    assert 400 == post.status_code
+    assert {'detail': "Registration already done"} == post.json()
+
+GOOD_DOMAINS = [
+    'liquidnode.local',
+    'however-bored-ill-write-tests.com',
+    'more.segments.please.org',
+    'some-dashes.auto',
+    'weird-tld.liquid',
+    'num8berx.zero',
+]
+
+BAD_DOMAINS = [
+    'under_score.com',
+    'other~things.xxx',
+    'dash-where-it.shouldnt-be',
+    'put+both-patches.cn',
+]
+
+@pytest.mark.parametrize("key,vals,broken_vals", [
+    ("lan", [LAN1, LAN2], [LAN_BROKEN]),
+    ("wan", [WAN1, WAN2], [WAN_BROKEN]),
+    ("ssh", [SSH1, SSH2], [SSH_BROKEN]),
+    ("domain", [{'domain': d} for d in GOOD_DOMAINS],
```

Comment:
How about changing `GOOD_DOMAINS` to this, so the fixture is shorter and fits on a single line?

```python
GOOD_DOMAINS = [{'domain': d} for d in [
    'liquidnode.local',
    'however-bored-ill-write-tests.com',
    'more.segments.please.org',
    'some-dashes.auto',
    'weird-tld.liquid',
    'num8berx.zero',
]]
```